### PR TITLE
Magnum Opus: Labor Rights

### PR DESCRIPTION
**Pull request by jwarwick**

### CODE REVIEW

File: test/clj/game_test/cards/events.clj [hunk 1]
```diff
@@ -1631,6 +1631,41 @@
     (click-prompt state :runner "HQ")
     (run-successful state)))
 
+(deftest labor-rights
+  ;; Labor Rights - trash 3 cards, shuffle 3 cards from heap->stack, draw 1 card, rfg Labor Rights
+  (testing "Basic behavior"
+    (do-game
+      (new-game {:runner {:deck ["Labor Rights" "Sure Gamble" "Lawyer Up" "Knifed"]}})
+      (take-credits state :corp)
+      (core/move state :runner (find-card "Sure Gamble" (:hand (get-runner))) :deck)
```
Comment (NoahTheDuke) [inline test/clj/game_test/cards/events.clj]: It should be possible to write these lines as `:deck ["Sure Gamble" "Lawyer Up" "Knifed"] :hand ["Labor Rights"]` in the `new-game` map.

File: src/clj/game/cards/events.clj [hunk 2]
```diff
@@ -1154,6 +1154,31 @@
...
 
+   "Labor Rights"
+   {:req (req (pos? (+ (count (:deck runner)) (count (:discard runner)))))
+    :effect (req (let [mill-count (min 3 (count (:deck runner)))]
+                   (mill state :runner :runner mill-count)
+                   (system-msg state :runner (str "trashes the top " mill-count (pluralize " card" mill-count) " of their Stack"))
+                   (let [heap-count (min 3 (count (get-in @state [:runner :discard])))]
+                     (continue-ability
+                       state side
+                       {:prompt (str "Choose " heap-count (pluralize " card" heap-count) " to shuffle into the stack")
```
Comment (Saintis) [inline src/clj/game/cards/events.clj]:
```suggestion
                       {:prompt (str "Choose " (quantify "card" heap-count) " to shuffle into the stack")
```
Or don't show the prompt when 3 or fewer cards in heap.

File: src/clj/game/cards/events.clj [hunk 3]
```diff
@@ -1154,6 +1154,31 @@
...
+                        :show-discard true
+                        :choices {:max heap-count
+                                  :all true
+                                  :not-self true
+                                  :req #(and (= (:side %) "Runner")
+                                             (in-discard? %))}
+                        :effect (req (doseq [c targets] (move state side c :deck))
+                                     (system-msg state :runner (str "moves " (join ", " (map :title targets)) " from their Heap to their Stack"))
```
Comment (Saintis) [inline src/clj/game/cards/events.clj]: I would stick to the text on the card (combining all messages into one).
```suggestion
                                     (system-msg state :runner (str "shuffles " (join ", " (map :title targets)) " from their Heap into their Stack, and draw 1 card"))
```

File: src/clj/game/cards/events.clj [hunk 4]
```diff
@@ -1154,6 +1154,31 @@
...
+                                     (shuffle! state :runner :deck)
+                                     (system-msg state :runner "shuffles their Stack")
```
Comment (Saintis) [inline src/clj/game/cards/events.clj]: And remove this
```suggestion
```

File: src/clj/game/cards/events.clj [hunk 5]
```diff
@@ -1154,6 +1154,31 @@
...
+                                     (draw state :runner 1)
+                                     (system-msg state :runner "draws 1 card")
```
Comment (Saintis) [inline src/clj/game/cards/events.clj]: And remove this
```suggestion
```

**Pull request merged**